### PR TITLE
ref(flags): remove frontend feature flag

### PR DESCRIPTION
**Pull request by michellewzhang**
if no feature flags to display on the chart, hide the feature flag legend on the graph:

SCR-20241202-mfse

### CODE REVIEW

**Review by cmanallen — APPROVED** [review]

File: static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx [hunk 1]
```diff
@@ -420,9 +419,7 @@ export function EventDetailsContent({
         </Fragment>
       ) : null}
       <EventContexts group={group} event={event} />
-      {hasFeatureFlagSection && (
-        <EventFeatureFlagList group={group} project={project} event={event} />
-      )}
+      {<EventFeatureFlagList group={group} project={project} event={event} />}
```
Comment (ryan953) [inline static/app/views/issueDetails/groupEventDetails/groupEventDetailsContent.tsx:422]:
```suggestion
      <EventFeatureFlagList group={group} project={project} event={event} />
```

**Review by ryan953 — APPROVED** [review]

File: static/app/views/issueDetails/streamline/eventGraph.tsx [hunk 2]
```diff
@@ -233,7 +232,7 @@ export function EventGraph({group, event, ...styleProps}: EventGraphProps) {
       seriesData.push(releaseSeries as BarChartSeries);
     }
 
-    if (flagSeries.markLine && hasFeatureFlagFeature) {
+    if (flagSeries.markLine && flagSeries.type === 'line') {
```
Comment (ryan953) [inline static/app/views/issueDetails/streamline/eventGraph.tsx:235]: where did this condition come from? the image in the description looks like it's a bar chart...

Comment (michellewzhang) [reply static/app/views/issueDetails/streamline/eventGraph.tsx:235]: the flag series (the lines showing the feature flag changes) is a line chart. the bar chart part is the event/user graph

Comment (michellewzhang) [reply static/app/views/issueDetails/streamline/eventGraph.tsx:235]: see comment here https://github.com/getsentry/sentry/blob/1cabfbca746112f1dee03b51e61ea1136d554a75/static/app/views/issueDetails/streamline/hooks/useFlagSeries.tsx#L83

Comment (ryan953) [reply static/app/views/issueDetails/streamline/eventGraph.tsx:235]: oh, i see the `line` thing is unset when we return early with:

```
return {
  seriesName: t('Feature Flags'),
  markLine: {},
  data: [],
};
```

Comment (michellewzhang) [reply static/app/views/issueDetails/streamline/eventGraph.tsx:235]: yup, so if the type isn't set as `line` we know there's no flag chart data. so we don't want to show the legend

**Review by ryan953 — APPROVED** [review]

**Review by aliu39 — APPROVED** [review]

**Pull request merged**